shared-bindings/adcbuffer/BufferedInput.c

```diff
@@ -0,0 +1,180 @@
+        { MP_QSTR_buffer, MP_ARG_OBJ | MP_ARG_REQUIRED },
+        { MP_QSTR_sample_rate, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 500000} },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    // Validate Pin
+    const mcu_pin_obj_t *pin = validate_obj_is_free_pin(args[ARG_pin].u_obj);
+
+    // Buffer Pointer defined and allocated by user
+    mp_buffer_info_t bufinfo;
+    mp_get_buffer_raise(args[ARG_buffer].u_obj, &bufinfo, MP_BUFFER_READ);
+
+    // signed or unsigned, byte per sample
+    bool signed_samples = bufinfo.typecode == 'b' || bufinfo.typecode == 'h';
+    uint8_t bytes_per_sample = 1;
+
+    // Bytes Per Sample
+    if (bufinfo.typecode == 'h' || bufinfo.typecode == 'H') {
+        bytes_per_sample = 2;
+    } else if (bufinfo.typecode != 'b' && bufinfo.typecode != 'B' && bufinfo.typecode != BYTEARRAY_TYPECODE) {
+        mp_raise_ValueError(translate("sample_source buffer must be a bytearray or array of type 'h', 'H', 'b' or 'B'"));
+    }
+
+    // Validate sample rate here
+    uint32_t sample_rate = (uint32_t)mp_arg_validate_int_range(args[ARG_sample_rate].u_int, 1, 500000, MP_QSTR_sample_rate);
+
+    // Create local object
+    adcbuffer_bufferedinput_obj_t *self = m_new_obj(adcbuffer_bufferedinput_obj_t);
+    self->base.type = &adcbuffer_bufferedinput_type;
+
+    // Call local intereface in ports/common-hal/adcbuffer
+    common_hal_adcbuffer_bufferedinput_construct(self,
+        pin,
+        ((uint8_t *)bufinfo.buf),
+        bufinfo.len,
+        bytes_per_sample,
+        signed_samples,
+        sample_rate
+        );
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|     def deinit(self) -> None:
+//|         """Turn off the BufferedInput and release the pin for other use."""
+//|         ...
+//|
+STATIC mp_obj_t adcbuffer_bufferedinput_deinit(mp_obj_t self_in) {
+    adcbuffer_bufferedinput_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    common_hal_adcbuffer_bufferedinput_deinit(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(adcbuffer_bufferedinput_deinit_obj, adcbuffer_bufferedinput_deinit);
+
+STATIC void check_for_deinit(adcbuffer_bufferedinput_obj_t *self) {
+    if (common_hal_adcbuffer_bufferedinput_deinited(self)) {
+        raise_deinited_error();
+    }
+}
+
+//|  Provided by context manager helper.
+//|
+//|     def __exit__(self) -> None:
+//|         """Automatically deinitializes the hardware when exiting a context. See
+//|         :ref:`lifetime-and-contextmanagers` for more info."""
+//|         ...
+//|
+STATIC mp_obj_t adcbuffer_bufferedinput___exit__(size_t n_args, const mp_obj_t *args) {
+    (void)n_args;
+    common_hal_adcbuffer_bufferedinput_deinit(args[0]);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(adcbuffer_bufferedinput___exit___obj, 4, 4, adcbuffer_bufferedinput___exit__);
+
+//|     value: None
```

Comment:
`value: None`: I don't think you mean that. This should be documentation for `readmultiple(...)`.

Reply:
Changed as suggested.